should work with Gromacs 5.x

MDPOW should be using Gromacs 5.1.x, which is now pretty stable. GromacsWrapper recently moved to Gromacs 5.x Becksteinlab/GromacsWrapper#26 so this should work with the latest GW.

Here is the traceback 

```
mdpow/tests/test_equilibration.py EEE

================================= ERRORS =================================
________ ERROR at setup of TestEquilibration.test_simulation_file ________

self = <test_equilibration.TestEquilibration object at 0x7f454d4ad550>

    def setup(self):
        self.basedir = os.getcwd()
        self.directory_name = tempfile.mkdtemp()
        print("basedir:  {}".format(self.basedir))
        print("Copying required files into:  {}".format(self.directory_name))
        files = ["benzene.itp","benzene.pdb"]
        for f in files:
            shutil.copy("mdpow/tests/benzene/{}".format(f),"{0}/{1}".format(self.directory_name,f))
            print("\tbenzene/{0} --> {1}/{0}".format(f,self.directory_name))
        os.chdir(self.directory_name)
        print("Changed directory:  {}".format(os.getcwd()))
        for k in self.sims.keys():
            W = self.sims[k](molecule="BNZ")
            W.topology(itp="benzene.itp")
            W.solvate(struct="benzene.pdb")
>           W.energy_minimize()

/home/ikenney/Library/python/MDPOW/mdpow/tests/test_equilibration.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/ikenney/Library/python/MDPOW/mdpow/equil.py:369: in energy_minimize
    params = gromacs.setup.energy_minimize(**kwargs)
/home/ikenney/Library/python/GromacsWrapper/gromacs/setup.py:684: in energy_minimize
    gromacs.mdrun_d(**mdrun_args)
/home/ikenney/Library/python/GromacsWrapper/gromacs/core.py:382: in __call__
    return self.run(*args,**kwargs)
/home/ikenney/Library/python/GromacsWrapper/gromacs/core.py:160: in run
    results, p = self._run_command(*_args, **_kwargs)
/home/ikenney/Library/python/GromacsWrapper/gromacs/core.py:587: in _run_command
    self.check_failure(result, command_string=p.command_string)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <gromacs.tools.Mdrun_d object at 0x7f454d9a2310>
result = (1, '', '                      :-) GROMACS - gmx, VERSION 5.1.1 (-:\n\n                            GROMACS is written ...CS\nwebsite at http://www.gromacs.org/Documentation/Errors\n-------------------------------------------------------\n')
msg = 'Gromacs tool failed\nCommand invocation: gmx mdrun_d -c em.pdb -deffnm em -stepout 10 -v'
command_string = 'gmx mdrun_d -c em.pdb -deffnm em -stepout 10 -v'

    def check_failure(self, result, msg='Gromacs tool failed', command_string=None):
        rc, out, err = result
        if not command_string is None:
            msg += '\nCommand invocation: ' + str(command_string)
        had_success = (rc == 0)
        if not had_success:
            gmxoutput = "\n".join([x for x in [out, err] if not x is None])
            m = re.search(self.gmxfatal_pattern, gmxoutput, re.VERBOSE | re.DOTALL)
            if m:
                formatted_message = ['GMX_FATAL  '+line for line in m.group('message').split('\n')]
                msg = "\n".join(\
                    [msg, "Gromacs command %(program_name)r fatal error message:" % m.groupdict()] +
                    formatted_message)
            if self.failuremode == 'raise':
>               raise GromacsError(rc, msg)
E               GromacsError: [Errno 1] Gromacs tool failed
E               Command invocation: gmx mdrun_d -c em.pdb -deffnm em -stepout 10 -v

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -50,7 +50,9 @@ install:
 script:
   # set up gromacs environment: picked up by GromacsWrapper
   - source $($HOME/get_gmx $GMX_VERSION)
-  - py.test -s --cov mdpow mdpow/tests
+  # NOTE: to see ALL Gromacs output for debugging on Travis CI,
+  #       disable py.test capturing with '-s'  
+  - py.test --cov mdpow mdpow/tests
 
 after_success:
   - codecov

--- a/.travis.yml
+++ b/.travis.yml
@@ -49,7 +49,7 @@ install:
 script:
   # set up gromacs environment: picked up by GromacsWrapper
   - source $($HOME/get_gmx $GMX_VERSION)
-  - py.test --cov mdpow mdpow/tests
+  - py.test -s --cov mdpow mdpow/tests
 
 after_success:
   - codecov

--- a/.travis.yml
+++ b/.travis.yml
@@ -12,7 +12,8 @@ branches:
 
 env:
     - GMX_VERSION=4.6.7
-#    - GMX_VERSION=5.1.2
+#- GMX_VERSION=5.1.2
+    - GMX_VERSION=2016
 
 # install python dependencies
 install:
@@ -44,13 +45,10 @@ install:
   # install MDPOW
   - pip install -e .
 
-  # source gromacs
-  - if [[ $GMX_VERSION == "4.6.7" ]]; then curl https://raw.githubusercontent.com/Becksteinlab/GromacsWrapper/develop/gromacs/templates/gromacswrapper_465.cfg -o ~/.gromacswrapper.cfg;
-    fi 
-  - source $($HOME/get_gmx $GMX_VERSION)
-  
 # run tests
-script: 
+script:
+  # set up gromacs environment: picked up by GromacsWrapper
+  - source $($HOME/get_gmx $GMX_VERSION)
   - py.test --cov mdpow mdpow/tests
 
 after_success:

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,10 +9,11 @@ branches:
     only:
         - master
         - develop
+        - testing-gromacs-5-travis
 
 env:
     - GMX_VERSION=4.6.7
-#- GMX_VERSION=5.1.2
+    - GMX_VERSION=5.1.2
     - GMX_VERSION=2016
 
 # install python dependencies

--- a/AUTHORS
+++ b/AUTHORS
@@ -19,3 +19,7 @@ their first commit. GitHub handle is optional.
 
 - Ian M Kenney (ianmkenney)
 
+2016
+----
+
+- Ryan Heilman (rhheilma)

--- a/CHANGES
+++ b/CHANGES
@@ -5,15 +5,22 @@ CHANGES for MDPOW
 Add summary of changes for each release. Use ISO dates. Reference
 GitHub issues numbers and PR numbers.
 
-2016-xx-xx    0.6.1
-orbeckst, iorga, ianmkenney
+2017-xx-xx    0.6.1
+orbeckst, iorga, ianmkenney, rhheilma
 
+* works with Gromacs 2016.x, 5.1.x, and 4.6.x (#36)
 * removed unused analysis.py and analysis/thermodynamics.py
 * supported water models: TIP4P, TIP3P, TIP5P, SPC, SPC/E, and
   new: M24 (modified TIP3P, #46)
-* fixed sc-alpha can only take integer powers (#71)
+* fixed sc-power can only take integer powers (#71)
 * use PyBOL for building the testing environment
-
+* removed run.MDrunnerSimple (#60)
+* used flexible water model (-DFLEXIBLE) for more robust energy
+  minimization
+* FEP failed with IndexError when lambdas where presented as a list
+  (#78)
+* EDR files are now output by default instead of XVG (#75)
+  
 
 2016-06-29    0.6.0
 orbeckst, ianmkenney

--- a/mdpow/templates/em_opls.mdp
+++ b/mdpow/templates/em_opls.mdp
@@ -2,7 +2,7 @@
 ; Template for energy minimization with OPLS-AA (part of POW)
 
 include                  = 
-define                   = 
+define                   = -DFLEXIBLE
 
 integrator               = steep
 

--- a/mdpow/tests/test_equilibration_script.py
+++ b/mdpow/tests/test_equilibration_script.py
@@ -29,8 +29,8 @@ class TestEquilibriumScript(object):
             try:
                 self._run_equil('water','benzene/')
                 self._new_structures()
-            except:
-                raise AssertionError('Equilibration simulations failed.')
+            except Exception as err:
+                raise AssertionError('Equilibration simulations failed with exception:\n{0}'.format(str(err)))
 
     def _new_structures(self):
         assert os.path.exists(

--- a/mdpow/tests/test_fep.py
+++ b/mdpow/tests/test_fep.py
@@ -16,7 +16,7 @@ class TestFEPschedule(object):
          'name': 'vdw',
          'sc_alpha': 0.5,
          'sc_power': 1,
-        'sc_sigma': 0.3},
+         'sc_sigma': 0.3},
         'Coulomb':
         {'couple_lambda0': 'vdw-q',
          'couple_lambda1': 'vdw',

--- a/mdpow/tests/test_fep_script.py
+++ b/mdpow/tests/test_fep_script.py
@@ -35,5 +35,7 @@ class TestFEPScript(object):
             try:
                 self._run_fep('water', 'benzene/')
             except:
-                raise AssertionError('FEP simulations failed.')
-            assert os.path.exists(os.path.join(self.tmpdir.name, 'benzene','FEP','water', 'VDW', '0000', 'md.edr'))
+                raise AssertionError('FEP simulations failed with exception:\n{0}'.format(str(err)))
+
+            assert os.path.exists(os.path.join(self.tmpdir.name,
+                                               'benzene', 'FEP', 'water', 'VDW', '0000', 'md.edr'))

--- a/mdpow/tests/testing_resources/states/base/runinput.yml
+++ b/mdpow/tests/testing_resources/states/base/runinput.yml
@@ -119,5 +119,8 @@ mdrun:
          # report progress verbosely (True is -v)
     nice: 19
          # nicel level
-    maxthreads: None
-         # maximum number of threads (-nt), 0 or None lets mdrun automatically choose
+    maxthreads: 3
+         # maximum number of threads (-nt), 0 or None lets mdrun
+         # automatically choose
+         # For testing on travis, needs to be either 3 or 1.
+         # See https://github.com/Becksteinlab/MDPOW/pull/77

--- a/mdpow/tests/testing_resources/states/base/runinput.yml
+++ b/mdpow/tests/testing_resources/states/base/runinput.yml
@@ -22,7 +22,7 @@ setup:
          # coordinate file name (pdb, gro, ... anything that Gromacs can read)
     maxwarn: 0
          # maximum number of warnings tolerated by grompp during setup
-    gromacsoutput: True
+    gromacsoutput: False
          # True: print output from Gromacs command to screen (useful for debugging)
          # False: only show the MDPOW messages
 

--- a/mdpow/tests/testing_resources/states/base/runinput.yml
+++ b/mdpow/tests/testing_resources/states/base/runinput.yml
@@ -22,7 +22,7 @@ setup:
          # coordinate file name (pdb, gro, ... anything that Gromacs can read)
     maxwarn: 0
          # maximum number of warnings tolerated by grompp during setup
-    gromacsoutput: False
+    gromacsoutput: True
          # True: print output from Gromacs command to screen (useful for debugging)
          # False: only show the MDPOW messages
 

--- a/mdpow/tests/testing_resources/states/base/runinput.yml
+++ b/mdpow/tests/testing_resources/states/base/runinput.yml
@@ -38,7 +38,7 @@ MD_relaxed:
     runlocal: True
          # True: launch Gromacs mdrun and wait for it
          # False: produce queuing system scripts and stop
-    runtime: 2
+    runtime: 1
          # run time in ps
     mdp: NPT_opls.mdp
          #  run parameter file for Gromacs;
@@ -49,7 +49,7 @@ MD_relaxed:
 MD_NPT:
     qscript: local.sh
          # queuing system scripts to produce
-    runtime: 10
+    runtime: 1
          # simulation run time in ps [50 ns is generous, probably can be shorter]
     runlocal: True
          # True: launch Gromacs mdrun and wait for it
@@ -65,7 +65,7 @@ FEP:
          # BAR or TI
     qscript: local.sh
          # queuing system scripts to produce
-    runtime: 5
+    runtime: 1
          # run time in ps for each free energy window
     runlocal: True
          # True: launch Gromacs mdrun and run all windows in a serial fashion

--- a/mdpow/tests/testing_resources/states/md_npt/runinput.yml
+++ b/mdpow/tests/testing_resources/states/md_npt/runinput.yml
@@ -119,5 +119,8 @@ mdrun:
          # report progress verbosely (True is -v)
     nice: 19
          # nicel level
-    maxthreads: None
-         # maximum number of threads (-nt), 0 or None lets mdrun automatically choose
+    maxthreads: 3
+         # maximum number of threads (-nt), 0 or None lets mdrun
+         # automatically choose
+         # For testing on travis, needs to be either 3 or 1.
+         # See https://github.com/Becksteinlab/MDPOW/pull/77

--- a/mdpow/tests/testing_resources/states/md_npt/runinput.yml
+++ b/mdpow/tests/testing_resources/states/md_npt/runinput.yml
@@ -22,7 +22,7 @@ setup:
          # coordinate file name (pdb, gro, ... anything that Gromacs can read)
     maxwarn: 0
          # maximum number of warnings tolerated by grompp during setup
-    gromacsoutput: True
+    gromacsoutput: False
          # True: print output from Gromacs command to screen (useful for debugging)
          # False: only show the MDPOW messages
 

--- a/mdpow/tests/testing_resources/states/md_npt/runinput.yml
+++ b/mdpow/tests/testing_resources/states/md_npt/runinput.yml
@@ -22,7 +22,7 @@ setup:
          # coordinate file name (pdb, gro, ... anything that Gromacs can read)
     maxwarn: 0
          # maximum number of warnings tolerated by grompp during setup
-    gromacsoutput: False
+    gromacsoutput: True
          # True: print output from Gromacs command to screen (useful for debugging)
          # False: only show the MDPOW messages
 

